Backend/app/controllers/partner.controller.js

```diff
@@ -1,31 +1,47 @@
-const Partner = require('../models/partner.model');
-const logger = require('../logger');
-
+const Partner = require("../models/partner.model");
+const logger = require("../logger");
 
 const partnerController = {
-    async getAllPartners(req, res) {
-        try {
-            const results = await Partner.listAll();
-            res.send({ partners: results });
-        } catch (err) {
-            console.error(err);
-            res.status(500);
-            res.send({ error: err });
-        }
-    },
-    async create(req, res) {
-        try {
-            logger.debug(req.body);
-            const new_partner = new Partner(req.body);
-            const partner_id = await new_partner.create();
-            logger.debug(partner_id);
-            res.json(partner_id);
-        } catch (err) {
-            console.error(err);
-            res.status(500);
-            res.send({ error: err });
-        }
+  async getAllPartners(req, res) {
+    try {
+      const results = await Partner.listAll();
+      res.send({ partners: results });
+    } catch (err) {
+      console.error(err);
+      res.status(500);
+      res.send({ error: err });
+    }
+  },
+  async create(req, res) {
+    try {
+      logger.debug(req.body);
+      const new_partner = new Partner(req.body);
+      const partner_id = await new_partner.create();
+      logger.debug(partner_id);
+      logger.info(
+        "Partner organization named %s added with id: %i",
+        new_partner.name,
+        partner_id
+      );
+      res.json(partner_id);
+    } catch (err) {
+      console.error(err);
+      res.status(500);
+      res.send({ error: err });
+    }
+  },
+  async getPartner(req, res) {
+    try {
+      const { name } = req.params;
+      logger.debug(req.params);
+      const partner_info = await Partner.getPartner(name);
+      res.send({ partnerInfo: partner_info });
+    } catch (err) {
+      console.error(err);
+      res.status(500);
+      res.send({ error: err });
     }
+  },
```

Comment:
Use the new error middleware, where `err` is passed to `next()`. See here https://github.com/uoftblueprint/homes-for-heroes/blob/main/Backend/app/controllers/customer.controller.js for reference.